dev/bots/test.dart

```diff
@@ -176,6 +178,55 @@ Future<bq.BigqueryApi> _getBigqueryApi() async {
   }
 }
 
+List<List<String>> _partitionToolTests() {
+  final List<File> pending = <File>[];
+  final String toolTestDir = path.join(toolRoot, 'test');
+  for (FileSystemEntity entity in Directory(toolTestDir).listSync(recursive: true)) {
+    if (entity is File && entity.path.endsWith('_test.dart')) {
+      pending.add(entity);
+    }
+  }
+  final List<String> groupA = <String>[];
+  final List<String> groupB = <String>[];
+  for (int i = 0; i < pending.length; i += 1) {
+    final String relativePath = path.relative(pending[i].path, from: toolRoot);
+    if (i.isEven) {
+      groupA.add(relativePath);
+    } else {
+      groupB.add(relativePath);
+    }
+  }
+  return <List<String>>[groupA, groupB];
+}
+
+Future<void> _runToolCoverage() async {
+  final List<List<String>> tests = _partitionToolTests();
+  await runCommand(
+    pub,
+    <String>['run', 'build_runner', 'build'],
+    workingDirectory: toolRoot,
+  );
+  // We run out of memory on CI if these are run together.
+  await runCommand(
+    dart,
+    <String>[path.join('tool', 'tool_coverage.dart'), '--']..addAll(tests.first),
+    workingDirectory: toolRoot,
+    environment: <String, String>{
+      'FLUTTER_ROOT': flutterRoot,
+      'SUBSHARD': 'A',
```

Comment:
Comment that these subshard names have to match what is in the .cirrus.yml. (do they?)